### DISCUSSION TITLE
Use credentials from `imagePullSecrets` in `trivy k8s` commands

### DISCUSSION THREAD

**Discussion by matheusfm** [aquasecurity/trivy#4742] [Ideas]
### Description

Hi all,

I would like to suggest that Trivy use the credentials from `imagePullSecrets` in the `trivy k8s --scanners vuln ...` commands, similar to how the Trivy Operator does: https://aquasecurity.github.io/trivy-operator/v0.14.1/docs/vulnerability-scanning/private-registries/#image-pull-secrets

### Target

Kubernetes

### Scanner

Vulnerability

**Comment by knqyf263** [comment]
@chen-keinan Isn't it supported now?

**Reply by chen-keinan** [reply]
@knqyf263  it is not supported for `trivy k8s ...`  subcommand. only simple global `user,password` array.

The operator use secrets in `trivy-system` namespace to manage `imagePullSecret with credential` for scan-job. (it is more suitable to job use-case)
I assume it can be added,  the only concern is to not expose cluster secrets.
